Documentation/ApiOverview/DependencyInjection/Index.rst

```diff
@@ -671,6 +671,70 @@ An :php:`Error` is thrown on missing dependency injection for
 
     Call to a member function methodName() on null
 
+.. index:: Dependency injection; Installation-wide configuration
+.. _dependency-injection-installation-wide:
+
+Installation-wide configuration
+-------------------------------
+
+..  versionadded:: 12.1
+
+One can set up a global service configuration for a project that can be used in
+multiple project-specific extensions. For example, this way you can alias an
+interface with a concrete implementation that can be used in several extensions.
+It is also possible to register project-specific :ref:`CLI commands
+<symfony-console-commands>` without requiring a project-specific extension.
+
+However, this only works - due to security restrictions - if TYPO3 is configured
+in a way that the project root is outside the document root, which is usually
+the case in Composer-based installations.
+
+The global service configuration files :file:`services.yaml` and
+:file:`services.php` are now read within the :file:`config/system/` path
+of a TYPO3 project in Composer-based installations.
+
+**Example:**
+
+You want to use the interface of the PHP package `stella-maris/clock` as a type
+hint for dependency injection in the service classes of your project's various
+extensions. Then the concrete implementation may change without touching your
+code. In this example, we use `lcobucci/clock` for the concrete implementation.
+
+..  code-block:: php
+    :caption: config/system/services.php
+
+    use Lcobucci\Clock\SystemClock;
```

Comment:
can we put this in a php file so it can have automatic checks run on?

Reply:
Hmm, the two classes/interfaces are not available, so a linting works but maybe not more. But I like to hold it together in this rst (and it is small) so I would prefer to let it inline.